cluster/src/main/java/com/hartwig/pipeline/stages/Stage.java

```diff
@@ -45,6 +46,10 @@ default S persistedOutput(final M metadata) {
         throw new UnsupportedOperationException(String.format("Stage [%s] does not support using persisted output.", namespace()));
     }
 
+    default List<AddDatatype> addDatatypes(final M metadata) {
```

Comment:
I think the method should just be called `datatypes` as it doesn't actually add them to anything.

On that topic, I think we can take this a small step further such that stages don't need to explictly add the datatypes in their output methods. This could happen in the StageRunner, which could add the the datatypes to the output.